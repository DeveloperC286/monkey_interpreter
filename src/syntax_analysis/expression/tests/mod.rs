use rstest::rstest;

#[rstest(
    code,
    snapshot_name,
    case("!", "test_syntax_analysis_for_prefix_expression_syntax_errors_case1"),
    case("-;", "test_syntax_analysis_for_prefix_expression_syntax_errors_case2")
)]
fn test_syntax_analysis_for_prefix_expression_syntax_errors(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name)
}

#[rstest(
    code,
    snapshot_name,
    case("!15", "test_syntax_analysis_for_prefix_expressions_case1")
)]
fn test_syntax_analysis_for_prefix_expressions(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name)
}
