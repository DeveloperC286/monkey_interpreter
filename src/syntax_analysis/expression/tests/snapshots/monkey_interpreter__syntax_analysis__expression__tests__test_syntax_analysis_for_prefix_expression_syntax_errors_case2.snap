---
source: src/syntax_analysis/expression/tests/mod.rs
expression: "crate::syntax_analysis::get_abstract_syntax_tree(crate::lexical_analysis::get_tokens(code))"
---
AbstractSyntaxTree {
    abstract_syntax_tree: [],
    syntax_parsing_errors: [
        "Syntax error : Do not know how to parse SemiColon as an expression.",
        "Syntax error : No right hand expression to prefix Minus.",
    ],
}
